### PR TITLE
CY-260 add new agents transfer cmd

### PR DESCRIPTION
**Pull request by adigrabow**

### CODE REVIEW

File: cloudify_cli/cli/helptexts.py [hunk 1]
```diff
@@ -244,3 +244,12 @@
                     'skips the first resource [default: 0]'
 DRY_RUN = 'If set, no actual operations will be performed. This ' \
           'only prints the executed tasks, without side effects'
+
+MANAGER_IP = 'The private IP of the new Cloudify-Manger'
```
Comment (mcouthon) [inline cloudify_cli/cli/helptexts.py]: Why the hyphen?

File: cloudify_cli/cli/helptexts.py [hunk 2]
```diff
@@ -244,3 +244,12 @@
...
+MANAGER_CERTIFICATE_PATH = 'A path to a file containing the SSL certificate' \
+                           ' of the new Cloudify Manager.' \
+                           ' the certificate is available on the Manager:' \
```
Comment (mcouthon) [inline cloudify_cli/cli/helptexts.py]: Upper case :)

File: cloudify_cli/cli/helptexts.py [hunk 3]
```diff
@@ -244,3 +244,12 @@
...
+                           ' /etc/cloudify/ssl/cloudify_internal_ca_cert.pem'
+MANAGER_REST_TOKEN = 'The REST token of the new Cloudify-Manager.' \
```
Comment (mcouthon) [inline cloudify_cli/cli/helptexts.py]: hyphen?

File: cloudify_cli/commands/agents.py [hunk 4]
```diff
@@ -133,7 +149,7 @@ def create_deployments_list(client, deployment_id, logger):
                 "Deployment '{0}' is not installed".format(deployment_id)
             return dep_list, error_msg
 
-        logger.info("Installing agent for deployment '{0}'"
+        logger.info("Running execution for deployment '{0}'"
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: Maybe pass the workflow ID here as well, and mention it in the log

File: cloudify_cli/commands/agents.py [hunk 5]
```diff
@@ -145,13 +161,14 @@ def create_deployments_list(client, deployment_id, logger):
             error_msg = 'There are no deployments installed'
             return dep_list, error_msg
 
-        logger.info('Installing agents for all installed deployments')
+        logger.info('Running execution for all installed deployments')
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: Ditto

File: cloudify_cli/commands/agents.py [hunk 6]
```diff
@@ -225,3 +242,74 @@ def worker(dep_id):
         ))
 
         raise SuppressedCloudifyCliError()
+
+
+@agents.command(name='transfer',
+                short_help='Configure agents to work with a new'
+                           ' Cloudify-Manager [Manager only]')
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: hyphen + manager only is lowercase

File: cloudify_cli/commands/agents.py [hunk 7]
```diff
@@ -225,3 +242,74 @@ def worker(dep_id):
         ))
 
         raise SuppressedCloudifyCliError()
+
+
+@agents.command(name='transfer',
+                short_help='Configure agents to work with a new'
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: I'd say something like "Transfer live agents to new ..."

File: cloudify_cli/commands/agents.py [hunk 8]
```diff
@@ -225,3 +242,74 @@ def worker(dep_id):
...
+             logger,
+             client,
+             all_tenants,
+             manager_ip,
+             manager_certificate,
+             manager_rest_token):
+    """Configure agents to work with a new Cloudify-Manager.
+
+    `DEPLOYMENT_ID` - The ID of the deployment you would like to
+    configure agents in.
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: in -> on

File: cloudify_cli/commands/agents.py [hunk 9]
```diff
@@ -225,3 +242,74 @@ def worker(dep_id):
...
+
+    """
+    with open(manager_certificate, 'r') as ssl_file:
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: Need to add verification here that the path exists before opening, and raise a helpful error

File: cloudify_cli/cli/helptexts.py (hunk 1 above)
Comment (mcouthon) [inline cloudify_cli/cli/helptexts.py]: Maybe add a little more info re which IP to use (e.g. "this is the IP through which the agents connect to the manager"

File: cloudify_cli/commands/agents.py [hunk 10]
```diff
@@ -79,6 +79,21 @@ def install(deployment_id,
     See Cloudify's documentation at http://docs.getcloudify.org for more
     information.
     """
+    get_deployments_and_run_worker(
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: worker -> workers

File: cloudify_cli/commands/agents.py [hunk 11]
```diff
@@ -225,3 +242,74 @@ def worker(dep_id):
...
+@cfy.options.all_tenants
+@cfy.pass_logger
+@cfy.pass_client()
+def validate(deployment_id,
+             include_logs,
+             tenant_name,
+             logger,
+             client,
+             all_tenants):
+    """Validates the connection between the Cloudify-Manager and the'
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: Why the extra '?

File: cloudify_cli/commands/agents.py [hunk 12]
```diff
@@ -225,3 +242,74 @@ def worker(dep_id):
...
+                ' Cloudify-Agents (installed on remote hosts).
+
+        `DEPLOYMENT_ID` - The ID of the deployment you would like to
+        validate agents for.
+
+        """
+    # params = {'manager_ip': manager_ip}
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: remove

File: cloudify_cli/commands/agents.py [hunk 13]
```diff
@@ -270,23 +272,32 @@ def transfer(deployment_id,
...
-    configure agents in.
+    configure agents on.
 
     """
-    with open(manager_certificate, 'r') as ssl_file:
-        manager_certificate = ssl_file.read()
+    if not os.path.exist(manager_certificate):
+        logger.error("Manager's SSL certificate file does not exist: {0}".
+                     format(manager_certificate))
+        raise IOError
```
Comment (mcouthon) [inline cloudify_cli/commands/agents.py]: Raise nice errors with nice output

File: cloudify_cli/commands/agents.py [hunk 14]
```diff
@@ -225,3 +244,82 @@ def worker(dep_id):
...
+def transfer(deployment_id,
+             include_logs,
+             tenant_name,
+             logger,
+             client,
+             all_tenants,
+             manager_ip,
+             manager_certificate,
+             manager_rest_token):
+    """Configure agents to work with a new Cloudify-Manager.
```
Comment (tehasdf) [inline cloudify_cli/commands/agents.py]: also a hyphen here, change to `Cloudify Manager` please

File: cloudify_cli/commands/agents.py [hunk 15]
```diff
@@ -225,3 +244,82 @@ def worker(dep_id):
...
+
+    `DEPLOYMENT_ID` - The ID of the deployment you would like to
+    configure agents on.
+
+    """
+    if not os.path.exist(manager_certificate):
```
Comment (tehasdf) [inline cloudify_cli/commands/agents.py]: `exists` (.exist isn't a function)

**Pull request merged**